fix(table): pagination, toolbar toolips; (card): tooltip

erzhan-temir-mamyrov opened the pull request:
**Summary**

Changed approach to previous issues:
- #3682 Card tooltip - add throttle to scroll event listener (initial render of a tooltip caused micro-scrolling event, therefore tooltip closed itself, throttle for 10ms prevents this)
- #3278 Pagination in table - decrease padding slightly between `608px` and `500px` instead of hiding items per page block
- Carbon tooltips in table - change direction from `top` to `bottom` in table toolbar (due to `overflow: hidden`)

**Change List (commits, features, bugs, etc)**

- Card tooltip: add `setTimeout` to add scroll event listener
- Table pagination: update css styles
- Tooltips in table toolbar: add props to change direction to `bottom`

**Acceptance Test (how to verify the PR)**

**Card tooltip**
- Go to [this story](https://deploy-preview-3690--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-card-card--with-ellipsed-title-tooltip-external-tooltip)
- Resize the window vertically in order to display vertical scrollbar
- Click on card or subtitle to show tooltip
- Scroll windows vertically
- Verify that tooltip was removed after the scrolling event

**Table pagination**
- Go to [this story](https://deploy-preview-3690--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-pagination)
- Resize the window horizontally till `500px`
- Verify that icon buttons to change pages do not overflow container

**Table toolbar tooltips**
- Go to [this story](https://deploy-preview-3690--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--playground)
- Set `hasFilter: true`, `hasSearch: true`, `hasAggregations: true` or any other button icons in tooltip
- Hover over icons in toolbar
- Verify that tooltips have direction `bottom`

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

<!-- For reviewers: do not remove -->

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

davidicus commented on packages/react/src/components/Card/CardTitle.jsx:
```diff
@@ -101,32 +107,81 @@ export const CardTitle = (
+    if (tooltipsState.title) {
+      window.addEventListener('scroll', handleTitleScroll);
+    } else {
+      window.removeEventListener('scroll', handleTitleScroll);
```
We need to return a function that removes these listeners from the useEffect

davidicus commented on packages/react/src/components/Card/CardTitle.jsx:
```diff
@@ -101,32 +107,81 @@ export const CardTitle = (
+  /* istanbul ignore next */
+  useEffect(() => {
+    if (tooltipsState.info) {
+      window.addEventListener('scroll', handleInfoScroll);
```
same comment about cleaning this up

davidicus reviewed (commented):
This code looks good and 2 of the 3 acceptance test worked as expected. The first one however does not close on scroll. The tooltip seems to stay with the content it is attached to so that seems acceptable but is this what the desired effect was supposed to be?

davidicus commented:
This is the acceptance test that does not work as stated. 

**Card tooltip**
- Go to [this story](https://deploy-preview-3690--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-card-card--with-ellipsed-title-tooltip-external-tooltip)
- Resize the window vertically in order to display vertical scrollbar
- Click on card or subtitle to show tooltip
- Scroll windows vertically
- Verify that tooltip was removed after the scrolling event

erzhan-temir-mamyrov commented:
@davidicus Thanks for commenting on this! I replaced throttling with setTimeout, it seems to do the trick. Also, I added cleanup functions for timers and event listeners.
